src/App.tsx

```diff
@@ -1,24 +1,190 @@
+
+    if (clickedTodo) {
+      updateTodos({
+        id: todoId,
+        completed: !clickedTodo.completed,
+      })
+        .then((response) => {
+          setTodos(prevTodos => {
+            return prevTodos.map((todo) => {
+              if (todo.id === response.id) {
+                return response as Todo;
+              }
+
+              return todo;
+            });
+          });
+        })
+        .catch(() => setErrorMesssage(Errors.update));
+    }
+  };
+
+  useEffect(() => {
+    setIsLoading(true);
+    getTodos(USER_ID)
+      .then((fetchedTodos) => {
+        setTodos(fetchedTodos);
+      })
+      .catch(() => setErrorMesssage(Errors.load))
+      .finally(() => {
+        setIsLoading(false);
+      });
+  }, []);
+
+  const filteredTodos = useMemo(() => [...todos].filter(({ completed }) => {
+    if (filter === Filters.All) {
+      return true;
+    }
+
+    return (filter === Filters.Active)
+      ? !completed
+      : completed;
+  }), [todos, filter]);
+
+  const removeTodo = (todoId: number) => {
+    setIsLoading(true);
+    setLoadingItems(current => [...current, todoId]);
+
+    deleteTodo(todoId)
+      .then(() => setTodos(current => current
+        .filter(todo => todo.id !== todoId)))
+      .catch(() => setErrorMesssage(Errors.delete))
+      .finally(() => {
+        setIsLoading(false);
+        // eslint-disable-next-line max-len
+        setLoadingItems(current => current.filter((id: number) => id !== todoId));
+      });
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!title.trim()) {
+      return setErrorMesssage(Errors.title);
+    }
+
+    const newTodo = {
+      id: 0,
+      userId: USER_ID,
+      title: title.trim(),
+      completed: false,
+    };
+
+    setTempTodo(newTodo);
+
+    setIsDisable(true);
+
+    return addTodo(newTodo)
+      .then((resultTodo:Todo) => {
+        setTodos(current => [...current, resultTodo]);
+        setTitle('');
+      })
+      .catch(() => setErrorMesssage(Errors.add))
+      .finally(() => {
+        setIsDisable(false);
+        setTempTodo(null);
+      });
+  };
+
+  const clearCompletedTodos = () => {
+    // eslint-disable-next-line max-len
+    const completedTodoIds = todos.filter(({ completed }) => completed).map(({ id }) => id);
+
+    completedTodoIds.forEach((id) => removeTodo(id));
+  };
+
+  const handleToggleAll = () => {
+    const areAllTodosCompleted = todos.every((todo) => todo.completed);
+    const updatedTodos = todos.map((todo) => ({
+      ...todo,
+      completed: !areAllTodosCompleted,
+    }));
+
+    setTodos((prevTodos) => {
+      updatedTodos.forEach((updatedTodo) => {
+        if (updatedTodo.completed !== prevTodos.find((t) => t.id
+        === updatedTodo.id)?.completed) {
+          updateTodos(updatedTodo)
+            .catch(() => setErrorMesssage(Errors.update));
+        }
+      });
```

Comment:
I'm not sure if this is the best solution here. This might be the reasone why tests:
```
 3) should become not active after toggling a todo
 4) should become not active
 5) should not be active
 6) should not become active after toggling a todo
 ```
 are failing.